client/grpc_query.go

```diff
@@ -0,0 +1,43 @@
+package client
+
+import (
+	gocontext "context"
+	"fmt"
+
+	gogogrpc "github.com/gogo/protobuf/grpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/encoding"
+	"google.golang.org/grpc/encoding/proto"
+)
+
+// QueryConn returns a new grpc ClientConn for making grpc query calls that
+// get routed to the node's ABCI query handler
+func (ctx Context) QueryConn() gogogrpc.ClientConn {
+	return cliQueryConn{ctx}
+}
+
+type cliQueryConn struct {
+	ctx Context
+}
+
+var _ gogogrpc.ClientConn = cliQueryConn{}
+
+var protoCodec = encoding.GetCodec(proto.Name)
+
+// Invoke implements the grpc ClientConn.Invoke method
+func (c cliQueryConn) Invoke(_ gocontext.Context, method string, args, reply interface{}, _ ...grpc.CallOption) error {
```

Comment:
Out of curiosity, `cliQueryConn` is just a wrapper around `ctx`, why not simply have `Context` implement `gogogrpc.ClientConn`?

Reply:
Oh l think that could work! 👍